docs/agent.md

```diff
@@ -0,0 +1,59 @@
+# agent (Private Beta)
+
+A hybrid agent has the capability to scan a resource within your Virtual Private Cloud (VPC) environment based on the inputs provided through the HCP UI.
+The `agent` command is used to fetch scan jobs for resources configured in [HCP](https://portal.cloud.hashicorp.com/) and execute them in your VPC. It will scan any available jobs, otherwise it will sleep and wait for a job to be queued in HCP.
+
+## Setting Up
+
+In order to run `agent` in your environment there are 3 configuration steps required in HCP and your local environment.
+
+### Create a Service Principal
+
+You will need to log into the HCP web portal to create your Service Principals. You can follow the HCP docs to create your [Project Service Principals](https://developer.hashicorp.com/hcp/docs/hcp/admin/iam/service-principals#project-level-service-principals-1). You will need to create the Service Principals at Admin level, more information about this level contained in [these](https://developer.hashicorp.com/hcp/docs/hcp/admin/iam/users#project-role) HCP docs. You will also need to generate a [service principal key](https://developer.hashicorp.com/hcp/docs/hcp/admin/iam/service-principals#generate-a-service-principal-key) **that gives you a client id and client secret that you will use in the next few steps**.
+
+### Create an agent pool in the HCP Portal
+
+> **_Note:_** Before you create a agent, there is currently no way to delete agents via the UI
+
+From the settings page, there will be an `Agent` tab. Select and follow the instructions to create a new Agent Pool. Alternatively attempting to on board a new agent data source will direct you to create an Agent Pool if you have not already.
+
+### Configure your Local Environment
+
+To run agent you will need your client id and client secret from the service principal you just created, as well as your project and agent pool id, which will be provided to you during your agent creation flow.
+
+To run the agent, the following environment variables must be set:
+
+```bash
+export HCP_PROJECT_ID=<HCP_PROJECT_ID>
+export HCP_CLIENT_ID=<HCP_CLIENT_ID>
+export HCP_CLIENT_SECRET=<HCP_CLIENT_SECRET>
+
+# The Agent Pool Id that was provided after creation
```

Comment:
nit: `after agent pool creation flow via UI` (just trying to be consistent in terminology)